include/clover_field_order.h

```diff
@@ -881,20 +1026,36 @@ namespace quda {
   } // namespace clover
 
   // Use traits to reduce the template explosion
-  template<typename Float,int N=72, bool add_rho=false> struct clover_mapper { };
+  template <typename Float, int N = 72, bool add_rho = false, bool enable_reconstruct = clover::reconstruct()>
+  struct clover_mapper {
+  };
 
   // double precision uses Float2
-  template<int N, bool add_rho> struct clover_mapper<double,N,add_rho> { typedef clover::FloatNOrder<double, N, 2, add_rho> type; };
+  template <int N, bool add_rho, bool enable_reconstruct> struct clover_mapper<double, N, add_rho, enable_reconstruct> {
+    using type = clover::FloatNOrder<double, N, 2, add_rho, enable_reconstruct>;
+  };
 
   // single precision uses Float4
-  template<int N, bool add_rho> struct clover_mapper<float,N,add_rho> { typedef clover::FloatNOrder<float, N, 4, add_rho> type; };
+  template <int N, bool add_rho, bool enable_reconstruct> struct clover_mapper<float, N, add_rho, enable_reconstruct> {
+    using type = clover::FloatNOrder<float, N, 4, add_rho, enable_reconstruct>;
+  };
+
+#ifdef FLOAT8
+#define N8 8
+#else
+#define N8 4
```

Comment:
Can this be changed to something like

```C++
#ifdef FLOAT8
 #define FLOATN 8
 #else
 #define FLOATN 4
```
with the appropriate substations to `N8` below?

Reply:
84359ab (and I applied the same cleanup to gauge_field_order.h and color_spinor_field_order.h)